utkdigitalinitiatives/basic-solr-config
# 4.x - Solr search exclusions and Fedora Generic Search multi-threading

## What does this pull request do?

This PR provides the following updates:
1. Solr search exclusions (e.g. undescribed collection and islandora namespace objects)
2. Configuration files for multi-threading the fedoragsearch webapp
## What's new?

Lots. There are updates to the README, new directories for the various configuration files, modifications to those configuration files, as well as minor updates to the islandora_transforms.
## How should this be tested?

The truth is that some of the changes are already in place on dev and production; this is the case for many of the islandora_transforms tweaks and the fedoragsearch performance updates. 

Testing /reviewing the changes made to solr-conf/solrconfig.xml is possible. These changes have not been deployed to production.

**note:** this assumes that you are using a minimal vm and do not have islandora_collection_search installed.
- start your vm
- visit ~~`http://localhost:8080/solr/collection1/select?q=%3a&lf=PID&wt=xml&indent=true`~~ `http://localhost:8080/solr/collection1/select?q=*%3A*&fl=PID&wt=xml&indent=true`
- this Solr query returns the first 10 items in the index - it should look something like this:

``` xml
<response>

<lst name="responseHeader">
  <int name="status">0</int>
  <int name="QTime">0</int>
  <lst name="params">
    <str name="q">*:*</str>
    <str name="indent">true</str>
    <str name="fl">PID</str>
    <str name="wt">xml</str>
  </lst>
</lst>
<result name="response" numFound="46" start="0">
  <doc>
    <str name="PID">islandora:bookCollection</str></doc>
  <doc>
    <str name="PID">islandora:compound_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_disk_image_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_pdf_collection</str></doc>
  <doc>
    <str name="PID">islandora:video_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_web_archive_collection</str></doc>
...[snip]...
```

To **exclude** the first two results, `islandora:bookCollection` and `islandora:compound_collection`
- use your favorite text editor to modify `/usr/local/solr/collection1/conf/solrconfig.xml`; e.g.
  - `$ sudo vim /usr/local/solr/collection1/conf/solrconfig.xml`
- at approximately line 800, add the following and save the file

``` xml
<lst name="appends">
  <str name="fq">-PID:islandora\:bookCollection -PID:islandora\:compound_collection</str>
</lst>
```
- restart Tomcat; e.g. `$ sudo service tomcat7 restart`
- after Tomcat has restarted, revisit the previous link ~~`http://localhost:8080/solr/collection1/select?q=%3a&lf=PID&wt=xml&indent=true`~~ `http://localhost:8080/solr/collection1/select?q=*%3A*&fl=PID&wt=xml&indent=true`. The PIDs you chose to exclude should not be in the query results.
## Additional Notes
## Interested Parties

@pc37utn @DonRichards 

**edit:** removed a Solr query that got copied incorrectly.
**edit:** fixed garbled Solr query URLs. oops. :/
**edit:** added note re `sudo` and modifying solrconfig.xml
